src/physics/thermal_conduction.cpp

```diff
@@ -130,47 +162,106 @@ void ThermalConduction::completeSetup()
     M_form_->Assemble(0);  // keep sparsity pattern of M and K the same
     M_form_->Finalize();
 
-    M_mat_.reset(M_form_->ParallelAssemble());
-
-    // Make the time integration operator and set the appropriate matrices
-    dyn_oper_ = std::make_unique<DynamicConductionOperator>(temperature_.space(), *dyn_M_params_, *dyn_T_params_, bcs_);
-    dyn_oper_->setMatrices(M_mat_.get(), K_mat_.get());
-    dyn_oper_->setLoadVector(rhs_.get());
-
-    ode_solver_->Init(*dyn_oper_);
-  }
-}
-
-void ThermalConduction::quasiStaticSolve()
-{
-  // Apply the boundary conditions
-  *bc_rhs_ = *rhs_;
-  for (auto& bc : bcs_.essentials()) {
-    bc.apply(*K_mat_, *bc_rhs_, temperature_, time_);
+    M_.reset(M_form_->ParallelAssemble());
+
+    residual_ = StdFunctionOperator(
+        temperature_.space().TrueVSize(),
+        [this](const mfem::Vector& du_dt, mfem::Vector& r) {
+          r = (*M_) * du_dt + (*K_) * (u_ + dt_ * du_dt);
+          r.SetSubVector(bcs_.allEssentialDofs(), 0.0);
+        },
+
+        [this](const mfem::Vector & /*du_dt*/) -> mfem::Operator& {
+          if (dt_ != previous_dt_) {
+            J_.reset(mfem::Add(1.0, *M_, dt_, *K_));
+            bcs_.eliminateAllEssentialDofsFromMatrix(*J_);
+          }
+          return *J_;
+        });
+
+    ode_ = FirstOrderODE(temperature_.trueVec().Size(), [=](const double t, const double dt, const mfem::Vector& u,
```

Comment:
Can we capture `[this]` here for clarity?

Reply:
good call, I missed this one